### ISSUE TITLE
Nushell compatibility

### ISSUE THREAD

**Issue by darshanCommits** [Vladimir-csp/uwsm#79]
`uwsm check` subcommands returns 1/0, which works for zsh/bash but breaks for nushell.

Making `uwsm check may-start/is-active` true/false would work for both scenario.

true/false.

tested command : 
`if (($env.TERM == "linux") and ((tty) == "/dev/tty1") and (uwsm check may-start)) {`
`   ^uwsm start -S hyprland.desktop`
`}`

**Comment by darshanCommits** [comment]
or maybe just document this since that might be more work.

```
def uwsm_may_start [] {
    let may_start = (uwsm check may-start | complete)
    $may_start.exit_code == 0
}

def uwsm_is_active [] {
    let may_start = (uwsm check is-active | complete)
    $may_start.exit_code == 0
}
```

**Comment by hardfau1t** [comment]
Another way to use is 
```nu
try {
    uwsm check may-start
    uwsm select
    uwsm start default 
}
```
since any one of the failure will stop the execution

**Comment by hardfau1t** [comment]
Bigger issue for me is, I use nushell as login shell, it sets bunch of environment variable, which are used by some shortcuts from hyprland. I am unable to figure out how to pass these variables to Hyprland(or systemd environments).

**Comment by DaRacci** [comment]
> Bigger issue for me is, I use nushell as login shell, it sets bunch of environment variable, which are used by some shortcuts from hyprland. I am unable to figure out how to pass these variables to Hyprland(or systemd environments).

Yeah I've just ran into this too, wondering if anyone knows a workaround?

**Comment by Vladimir-csp** [comment]
Since v0.21.2 uwsm picks up environment state `uwsm start` was executed in. Are those variables exported at this point?

**Comment by DaRacci** [comment]
i figured out my issue was more so a me issue, i didn't have hm generating a .profile file after getting that made and pointing to source the files it worked but beforehand, it didn't seem to be picking anything up since so I'm assuming for some reason it wasn't executed inside a login shell at that point (Maybe the solution could be spawning a login shell manually and taking its environment too?)

Also I'm using v0.21.3

My relevant configurations can be found here:
https://github.com/DaRacci/nix-config/blob/master/modules/nixos/shared/display-manager.nix
https://github.com/DaRacci/nix-config/blob/master/hosts/shared/optional/hyprland.nix
https://github.com/DaRacci/nix-config/blob/master/home/shared/desktop/hyprland/default.nix
https://github.com/DaRacci/nix-config/blob/master/home/shared/desktop/hyprland/nvidia.nix
https://github.com/DaRacci/nix-config/blob/69e0917b5b1d400522ce63135d3f16a30207db7b/home/racci/features/desktop/hyprland/default.nix#L55-L65
https://github.com/DaRacci/nix-config/blob/69e0917b5b1d400522ce63135d3f16a30207db7b/home/shared/features/cli/nushell.nix#L11-L26
